token-swap/program/fuzz/fuzz_targets/instructions.rs

```diff
@@ -0,0 +1,258 @@
+                .entry(token_a_id)
+                .or_insert(token_swap.create_token_a_account(INITIAL_USER_TOKEN_A_AMOUNT));
+            let mut token_b_account = token_b_accounts
+                .entry(token_b_id)
+                .or_insert(token_swap.create_token_b_account(INITIAL_USER_TOKEN_B_AMOUNT));
+            match trade_direction {
+                TradeDirection::AtoB => {
+                    token_swap.swap_a_to_b(&mut token_a_account, &mut token_b_account, instruction)
+                }
+                TradeDirection::BtoA => {
+                    token_swap.swap_b_to_a(&mut token_b_account, &mut token_a_account, instruction)
+                }
+            }
+        }
+        FuzzInstruction::Deposit {
+            token_a_id,
+            token_b_id,
+            pool_token_id,
+            instruction,
+        } => {
+            let mut token_a_account = token_a_accounts
+                .entry(token_a_id)
+                .or_insert(token_swap.create_token_a_account(INITIAL_USER_TOKEN_A_AMOUNT));
+            let mut token_b_account = token_b_accounts
+                .entry(token_b_id)
+                .or_insert(token_swap.create_token_b_account(INITIAL_USER_TOKEN_B_AMOUNT));
+            let mut pool_account = pool_accounts
+                .entry(pool_token_id)
+                .or_insert(token_swap.create_pool_account());
+            token_swap.deposit(
+                &mut token_a_account,
+                &mut token_b_account,
+                &mut pool_account,
+                instruction,
+            )
+        }
+        FuzzInstruction::Withdraw {
+            token_a_id,
+            token_b_id,
+            pool_token_id,
+            instruction,
+        } => {
+            let mut token_a_account = token_a_accounts
+                .entry(token_a_id)
+                .or_insert(token_swap.create_token_a_account(INITIAL_USER_TOKEN_A_AMOUNT));
+            let mut token_b_account = token_b_accounts
+                .entry(token_b_id)
+                .or_insert(token_swap.create_token_b_account(INITIAL_USER_TOKEN_B_AMOUNT));
+            let mut pool_account = pool_accounts
+                .entry(pool_token_id)
+                .or_insert(token_swap.create_pool_account());
+            token_swap.withdraw(
+                &mut pool_account,
+                &mut token_a_account,
+                &mut token_b_account,
+                instruction,
+            )
+        }
+    };
+    result
+        .map_err(|e| {
+            if e == SwapError::CalculationFailure.into() {
+            } else if e == SwapError::ConversionFailure.into() {
+            } else if e == SwapError::FeeCalculationFailure.into() {
+            } else if e == SwapError::ExceededSlippage.into() {
+            } else if e == SwapError::ZeroTradingTokens.into() {
+            } else if e == TokenError::InsufficientFunds.into() {
+            } else {
+                Err(e).unwrap()
+            }
```

Comment:
This allows certain errors to slide, since they don't mean anything truly wrong has happened.  Note that these checks do run the risk of overlap, since `ProgramError::Custom(u32)` can re-use the same value for `TokenError` and `SwapError`.  We should consider changing it to `ProgramError::Custom(Pubkey, u32)` to separate better